src/app/pages/basket/shopping-basket/shopping-basket.component.html

```diff
@@ -19,11 +19,20 @@ <h1 class="d-flex flex-wrap align-items-baseline">
     <!-- Error message -->
     <ish-error-message [error]="error"></ish-error-message>
 
-    <!-- Info messages -->
+    <!-- Basket Info messages -->
     <ish-basket-info></ish-basket-info>
 
     <!-- Validation messages-->
     <ish-basket-validation-results></ish-basket-validation-results>
+
+    <!-- Dynamic Promotion Message -->
+    <div *ngIf="basket.dynamicMessages?.length" class="alert alert-info">
```

Comment:
It could be possible, that the basket object from the input could be undefined. `basket?.dynamicMessages?.length` is better.